src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts

```diff
@@ -33,26 +33,73 @@ export function attachDebuggerHotkey(resourceLoader: WebAssemblyResourceLoader):
     if (evt.shiftKey && (evt.metaKey || evt.altKey) && evt.code === 'KeyD') {
       if (!debugBuild && !hasReferencedPdbs) {
         console.error('Cannot start debugging, because the application was not compiled with debugging enabled.');
+      } else if (navigator.userAgent.includes('Firefox')) {
+        launchDebugger(true);
       } else if (!currentBrowserIsChromeOrEdge) {
         console.error('Currently, only Microsoft Edge (80+), Google Chrome, or Chromium, are supported for debugging.');
       } else {
-        launchDebugger();
+        launchDebugger(false);
       }
     }
   });
 }
 
-function launchDebugger() {
+async function launchDebugger(isFirefox : boolean) {
   // The noopener flag is essential, because otherwise Chrome tracks the association with the
   // parent tab, and then when the parent tab pauses in the debugger, the child tab does so
   // too (even if it's since navigated to a different page). This means that the debugger
   // itself freezes, and not just the page being debugged.
   //
   // We have to construct a link element and simulate a click on it, because the more obvious
   // window.open(..., 'noopener') always opens a new window instead of a new tab.
-  const link = document.createElement('a');
-  link.href = `_framework/debug?url=${encodeURIComponent(location.href)}`;
-  link.target = '_blank';
-  link.rel = 'noopener noreferrer';
-  link.click();
+  if (isFirefox) {
```

Comment:
This is a lot of Firefox-specific code. Maybe it's worth to move it into a helper function so it's easier to read the logic here:
```js
if (isFirefox)
  launchFirefoxDebugger();
else{
 ...
}
```